### PR TITLE
Even more repository clean-up

### PR DESCRIPTION
**Pull request by kdmccormick**
Some more repo shuffling that I pulled out of https://github.com/edx/edx-drf-extensions/pull/91 in order to keep that PR purely API docs.

Changes:
* Add a generic `manage.py` file (cloned from https://github.com/edx/cookiecutter-django-app) so we can run `pytest .` and other commands that expect the repository to be like a Django project.
* Check quality in top-level modules (like `manage.py`)
* Speed up and simplify the Travis build a bit. Right now, the test builds are installing all development requirements before using `tox` (which wipes them away). Instead, install only the requirements to run `tox` before running it.

@jmbowman When you have a chance can you or someone else from arch-bom take a look? This isn't urgent at all.

cc @nedbat

### CODE REVIEW

File: requirements/tox.in [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Requirements for using tox.
```
Comment (jmbowman) [inline requirements/tox.in]: We usually create a `travis.in` file instead, that also includes codecov and tox-battery.

File: requirements/test.in [hunk 2]
```diff
@@ -3,7 +3,7 @@
 -c constraints.txt
 
 -r base.txt
-
+-r tox.txt
```
Comment (jmbowman) [inline requirements/test.in]: After adding the new requirements file, why does this one still need tox?  Normally this file represents what's installed for a test run inside of a tox virtualenv.

File: Makefile [hunk 3]
```diff
@@ -63,4 +69,4 @@ quality: style isort_check linting ## run all code quality checks in current env
 	@echo "Quality checking complete!"
 
 test-python: ## run unit tests within this environment only
-	python -Wd -m pytest
+	pytest
```
Comment (jmbowman) [inline Makefile]: Any particular reason for disabling deprecation warnings here?

**Comment by kdmccormick** [comment]
Since we moved the API doc tools work over to https://github.com/edx/api-doc-tools, I'm no longer working in this repo, so I'm going to close this PR. If anyone wants to re-open this PR and pull these changes into a new branch, they're welcome to.

**Pull request closed without merging**